app/invocation/invocation_action_card.tsx

```diff
@@ -0,0 +1,165 @@
+import pako from "pako";
+import React from "react";
+import format from "../format/format";
+import InvocationModel from "./invocation_model";
+import { build } from "../../proto/remote_execution_ts_proto";
+import rpcService from "../service/rpc_service";
+
+interface Props {
+  model: InvocationModel;
+  search: string;
+}
+
+interface State {
+  contents?: ArrayBuffer;
+  action_array?: Uint8Array;
+  action?: build.bazel.remote.execution.v2.Action;
+  command?: build.bazel.remote.execution.v2.Command;
+  error?: string;
+}
+
+export default class ActionCardComponent extends React.Component<Props, State> {
+  state: State = {};
+  componentDidMount() {
+    this.fetchAction();
+  }
+
+  fetchAction() {
+    // TODO: Replace localhost:1987 with the remote cache address and prefix the path with the instance name
+    let actionFile = "bytestream://localhost:1987/blobs/" + this.props.search.substring(14);
```

Comment:
What does this hardcoded 14 mean? (Also looks like it's used below during rendering)

Reply:
it's where within the window.location.search the action hash/size starts, i'll change it so that i substring when i call ActionCardComponent because i don't use the rest of the search string anywhere else

Reply:
I think you probably want to parse the search string in a less brittle way.

We actually already parse it at the root here: https://github.com/buildbuddy-io/buildbuddy/blob/c83e5a55058d3252cd76bffc8975d5a32d5940af/enterprise/app/root/root.tsx#L41

You can pass that down through the components to the action.

Then you can `get` just the parameter you want like this: https://github.com/buildbuddy-io/buildbuddy/blob/c83e5a55058d3252cd76bffc8975d5a32d5940af/app/invocation/invocation.tsx#L134